server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1383,7 +1393,13 @@ public MergeStats mergeStats() {
         if (engine == null) {
             return new MergeStats();
         }
-        return engine.getMergeStats();
+
+        final MergeStats mergeStats = engine.getMergeStats();
+        if (indexSettings.shouldCleanupUnreferencedFiles()) {
+            mergeStats.addTotalCleanUpStats(engine.unreferencedFileCleanupStats());
+        }
```

Comment:
To confirm, the unreferenced file cleanup stats will still show up with 0 values in merge stats (on or after the version we define) even if index settings don't allow the cleanup, correct?

If yes, can we add a test to assert this for both scenarios when the settings allow and don't allow the cleanup?

Reply:
Addressed the comments.